File: 222. Count Complete Tree Nodes/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def countNodes(self, root: TreeNode) -> int:
        if not root:
            return 0
        
        left = self.get_depth(root.left)
        right = self.get_depth(root.right)
        if left == right:
            return 2 ** left + self.countNodes(root.right)
        else:
            return 2 ** right + self.countNodes(root.left)
        
    def get_depth(self, root):
        if not root:
            return 0
        
        return 1 + self.get_depth(root.left)

File: 222. Count Complete Tree Nodes/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int countNodes(TreeNode root) {
        int count = 0;
        while (root != null) {
            if (getDepth(root.left) == getDepth(root.right)) {
                count += Math.pow(2, getDepth(root.left));
                root = root.right;
            } else {
                count += Math.pow(2, getDepth(root.right));
                root = root.left;
            }
        }
        return count;
    }
    
    public int getDepth(TreeNode root) {
        if (root == null) {
            return 0;
        } else {
            return 1 + getDepth(root.left);
        }
    }
}
